DOCS-3658 Adds Use Session Replay In Your Solutions/Support Workflows Guide + Troubleshooting Doc Page (#14445)
DataDog/documentation: content/en/real_user_monitoring/guide/session-replay-for-solutions.md

kind: guide
description: Learn how to adopt RUM & Session Replay in your Solutions or Support organization. 
further_reading:
- link: '/real_user_monitoring/connect_rum_and_traces/'
  tag: 'Documentation'
  text: 'Learn how to connect RUM with APM traces'
- link: '/real_user_monitoring/session_replay/'
  tag: 'Documentation'
  text: 'Learn about Session Replay'
- link: '/real_user_monitoring/session_replay/developer_tools'
  tag: 'Documentation'
  text: 'Learn about Browser Dev Tools'
---

## Overview

You can enable your technical solutions and support teams to better troubleshoot customer issues using [Session Replay][1]. With RUM & Session Replay, you can locate specific user sessions, observe user journeys, and access developer tools to see events, logs, errors, and attributes. 

This guide describes a workflow that organizations can replicate and use as an asset for solutions teams to integrate into their workflows.

{{< img src="real_user_monitoring/guide/session-replay/session-replay-recording.png" alt="Session Replay recording of a user session in the Shopist application" style="width:100%;">}}

## Assess user issues

> Let’s assume that a customer encounters an issue using Datadog. Your Technical Solutions team may use a support solution such as Zendesk or ServiceNow that creates a ticket when this customer reports that they cannot update or save a Synthetics multistep API test. 

Reviewer: ```suggestion
Assume that a customer encounters an issue using Datadog. Your Technical Solutions team may use a support solution, such as Zendesk or ServiceNow, that creates a ticket when this customer reports that they cannot update or save a Synthetics multistep API test. 
```

> The team may request more information from the customer (such as the specific test ID and a screen recording with the [Browser Dev Tools][2] open) in order to obtain some additional context into the customer's test not updating or saving. If no console errors were recorded, the team would not have any hints to start investigating the multistep API test's issue. 

Reviewer: ```suggestion
The team may request more information from the customer (such as the specific test ID and a screen recording with the [Browser Dev Tools][2] open) that may provide additional context for the customer's test not updating or saving. If no console errors were recorded, the team would not have any hints to start investigating the multistep API test's issue. 
```

The Technical Solutions team may try to understand the following questions:

- What is the exact error the customer is experiencing?
- Is the customer seeing an in-app notification that hints at a particular issue (such as a console error or an error message)?
- What buttons did the customer click on, and in what order? Did an unexpected action occur before the customer clicked on a button?

## Investigate the root cause

If there were a way to view the customer’s user journey in Datadog and see associated backend requests, the Technical Solutions team would have a better understanding of what may be causing this issue.

{{< img src="real_user_monitoring/guide/session-replay/apm-traces-in-session-replay.png" alt="An APM stack trace associated with a RUM view action" style="width:100%;">}}

With the APM integration, you can connect requests from your web application with corresponding backend traces to access APM trace data from a RUM event and uncover any backend errors in the **Errors** tab. 

For more information, see [Connect RUM and Traces][3].

## Watch user sessions in Session Replay

> The Technical Solutions team may have internal tools that connect a support platform, like Zendesk, with Datadog products, such as RUM & Session Replay. For example, if you click on a contextual link in Zendesk, it redirects you to the [RUM Explorer][4] and autofills the user ID in the search query. You can then filter for individual user sessions in the event list.

Reviewer: ```suggestion
The Technical Solutions team may have internal tools that connect a support platform, like Zendesk, with Datadog products, such as RUM & Session Replay. For example, a contextual link in Zendesk can redirects you to the [RUM Explorer][4] and autofill the user ID in the search query. You can then filter for individual user sessions in the event list.
```
